### PR TITLE
Codechecker fixes

### PR DESCRIPTION
**Pull request by lucaboesch**
Adhere to more recent Moodle Coding Guidelines.

### CODE REVIEW

File: classes/import/marksessions.php [hunk 1]
```diff
@@ -132,15 +132,15 @@ protected function get_column_data($row, $index) {
...
      * @param mod_attendance_structure $att The current assignment
+     * @param string $text The raw csv text.
      * @param string $encoding The encoding of the csv file.
      * @param string $delimiter The specified delimiter for the file.
      * @param string $importid The id of the csv import.
      * @param array $mappingdata The mapping data from the import form.
      * @param bool $useprogressbar Whether progress bar should be displayed, to avoid html output on CLI.
      */
-    public function __construct($text = null, $att, $encoding = null, $delimiter = null, $importid = 0,
+    public function __construct($att, $text = null, $encoding = null, $delimiter = null, $importid = 0,
```
Comment (danmarsden) [inline classes/import/marksessions.php:143]: did you change the order of these params on purpose? - this doen'st seem to be a codechecker change?

File: import/marksessions.php [hunk 2]
```diff
@@ -66,12 +66,12 @@
 
 $formparams = ['id' => $cm->id,
                'sessionid' => $pageparams->sessionid,
-               'grouptype' => $pageparams->grouptype];
+               'grouptype' => $pageparams->grouptype, ];
 $form = null;
 if (optional_param('needsconfirm', 0, PARAM_BOOL)) {
     $form = new \mod_attendance\form\import\marksessions($url->out(false), $formparams);
 } else if (optional_param('confirm', 0, PARAM_BOOL)) {
-    $importer = new \mod_attendance\import\marksessions(null, $att, null, null, $importid);
```
Comment (danmarsden) [inline import/marksessions.php:74]: looks like here'e the matching change for the function parameter order change - I think we should probably do this as it's own commit rather than bundling into the "codechecker fixes" if there's a good reason we should do this.

**Comment by lucaboesch** [comment]
Well spotted, @danmarsden.
I've added an individual commit for it.
Note that the catalyst automation seems still using actions/checkout@v3 instead of actions/checkout@v4.

Best,
Luca

**Comment by danmarsden** [comment]
cool thanks! - merged it in.
